registry/endpoint.go

```diff
@@ -260,3 +261,14 @@ HeaderLoop:
 
 	return RegistryInfo{}, fmt.Errorf("v2 registry endpoint returned status %d: %q", resp.StatusCode, http.StatusText(resp.StatusCode))
 }
+
+func (e *Endpoint) HTTPClient() *http.Client {
+	return &http.Client{
+		Transport: &http.Transport{
+			DisableKeepAlives: true,
+			Proxy:             http.ProxyFromEnvironment,
+			TLSClientConfig:   &tls.Config{InsecureSkipVerify: !e.IsSecure},
```

Comment:
@jlhawn I was wondering about just using newClient as well, however it seemed like a larger change.  The http.Client returned here matches what used to be used to be constructed in the locations that now call Endpoint.HTTPClient(), with the addition of TLSClientConfig.

I'm happy to switch to newClient, or just wait for #10265 to address this.  Preferences?